### PR TITLE
Update CircleCI image, Vault version, Ruby versions

### PR DESCRIPTION
**Pull request by swenson**
And take into account some slight behavior changes in newer Vault
versions.

Tests pass, but can be slightly flaky now:
https://app.circleci.com/pipelines/github/hashicorp/vault-ruby/136/workflows/f306166f-8d57-4043-893d-c291789c40ce

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -71,15 +62,15 @@ workflows:
...
+              vault-version: ["1.11.1", "1.10.5", "1.9.8"]
           name: test-ruby-<< matrix.ruby-version >>-vault-<< matrix.vault-version >>
       - build-release:
           requires:
             - test
           context: vault-gem-release
           filters:
             tags:
-              only: /^v[0-9]+\.[0-9]+\.[0-9]+.*/ 
+              only: /^v[0-9]+\.[0-9]+\.[0-9]+.*/
```
Comment (robmonte) [inline .circleci/config.yml:74]: I noticed this removed trailing space - just want to confirm this was intentional and not the IDE removing something that's potentially desired?

File: lib/vault/client.rb [hunk 2]
```diff
@@ -387,19 +388,20 @@ def success(response)
...
       if response.body && response.body.match("missing client token")
-        raise MissingTokenError
-      end
-
-      # Use the correct exception class
-      case response
-      when Net::HTTPPreconditionFailed
-        raise MissingRequiredStateError.new
-      when Net::HTTPClientError
+        # this behavior changed in Vault 1.10
```
Comment (calvn) [inline lib/vault/client.rb:391]: Can you elaborate on this a bit more? Someone who's not looking at this particular diff/PR might not know what this change is referring to.

Comment (swenson) [reply lib/vault/client.rb]: :+1:

**Review by benashz — APPROVED** [review]
LGTM!

**Comment by swenson** [comment]
Thanks. Going to merge this to get the build into better (if not perfect) shape

**Pull request merged**